retrieval_service/datastore/providers/firestore.py

```diff
@@ -42,41 +45,121 @@ def kind(cls):
+        create_vector_index = await asyncio.create_subprocess_exec(
+            "gcloud",
+            "alpha",
+            "firestore",
+            "indexes",
+            "composite",
+            "create",
+            f"--collection-group={collection_name}",
+            "--query-scope=COLLECTION",
+            '--field-config=field-path=embedding,vector-config={"dimension":768,"flat":"{}"}',
+            "--database=(default)",
+        )
+        await create_vector_index.wait()
+
+    async def __delete_vector_index(self, indices: list[str]):
+        # Check if the collection exists and deletes all indexes
+        for index in indices:
+            if index:
+                delete_vector_index = await asyncio.create_subprocess_exec(
+                    "gcloud",
+                    "alpha",
+                    "firestore",
+                    "indexes",
+                    "composite",
+                    "delete",
+                    index,
+                    "--database=(default)",
+                    "--quiet",  # Added to suppress delete warning
+                )
+                await delete_vector_index.wait()
+
     async def initialize_data(
         self,
         airports: list[models.Airport],
         amenities: list[models.Amenity],
         flights: list[models.Flight],
         policies: list[models.Policy],
     ) -> None:
-        async def delete_collections(collection_list: list[AsyncCollectionReference]):
-            # Checks if colelction exists and deletes all documents
-            delete_tasks = []
-            for collection_ref in collection_list:
-                collection_exists = collection_ref.limit(1).stream()
-                if not collection_exists:
-                    continue
-
-                docs = collection_ref.stream()
-                async for doc in docs:
-                    delete_tasks.append(asyncio.create_task(doc.reference.delete()))
-            asyncio.gather(*delete_tasks)
-
         # Check if the collections already exist; if so, delete collections
         airports_ref = self.__client.collection("airports")
         amenities_ref = self.__client.collection("amenities")
         flights_ref = self.__client.collection("flights")
         policies_ref = self.__client.collection("policies")
-        await delete_collections(
+        await self.__delete_collections(
             [airports_ref, amenities_ref, flights_ref, policies_ref]
         )
 
-        # initialize collections
+        # List indexes to check if the collections already exist; if so, delete collections
+        # Assign the index-id to the corresponding collection
+        indexes = await self.__list_vector_index()
+        if indexes != [""]:
+            collections = {"amenities": "", "policies": ""}
+            for line in indexes:
+                collection, index_id = await self.parse_index_info(line)
+                collections[collection] = index_id
```

Comment:
Still some room for improvement here. 

I think you should update `__list_vector_index` to return a dict where `collection_id` -> `index-id`. This means that `__list_vector_index` would call `self.parse_index_info` to parse the line into the right format, include placing the parsed results into a dict. 

The reason for this is that when you are returning a string, you still have additional parsing to do to get something that is useable. By returning the parsed version up front, you return a data type that can be immediately used. It also means that if later on their becomes an API we can call to get indexes, we can just update the 1 place (as opposed to having to update the code to use the API and remove the parse). 

It would simplify this section of code to this (which is considerably more readable):
```
        indexes = await self.__list_vector_index()
        amenities_ref = indexes.get("amenities", "")
        policies_ref = indexes.get("policies", "")
```

you might also want to consider renaming from `__list_vector_index` to `__get_indices` after this change as well.